src/workload_api/client.rs

```diff
@@ -0,0 +1,471 @@
+    /// there is a problem processing the response.
+    pub fn fetch_x509_context(&self) -> Result<X509Context, ClientError> {
+        let request = X509SVIDRequest::new();
+
+        let mut response = self
+            .client
+            .fetch_x509_svid_opt(&request, WorkloadApiClient::call_options()?)?;
+
+        let item = block_on(response.next());
+        let x509_context = WorkloadApiClient::process_x509_context(item)?;
+        Ok(x509_context)
+    }
+
+    /// Fetches a [`JwtSvid`] parsing the JWT token in the Workload API response, for the given audience and spiffe_id.
+    ///
+    /// # Arguments
+    ///
+    /// * `audience`  - A list of audiences to include in the JWT token. Cannot be empty nor contain only empty strings.
+    /// * `spiffe_id` - Optional [`SpiffeId`] for the token 'sub' claim. If not provided, the Workload API returns the
+    /// default identity.
+    ///
+    /// # Errors
+    ///
+    /// The function returns a variant of [`ClientError`] if there is en error connecting to the Workload API or
+    /// there is a problem processing the response.
+    ///
+    /// IMPORTANT: If there's no registration entries with the requested [`SpiffeId`] mapped to the calling workload,
+    /// it will return a [`ClientError::EmptyResponse`].
+    pub fn fetch_jwt_svid<T: AsRef<str> + ToString>(
+        &self,
+        audience: &[T],
+        spiffe_id: Option<&SpiffeId>,
+    ) -> Result<JwtSvid, ClientError> {
+        let response = self.fetch_jwt(audience, spiffe_id)?;
+        match response.svids.get(DEFAULT_SVID) {
+            Some(r) => Ok(JwtSvid::from_str(&r.svid)?),
+            None => Err(ClientError::EmptyResponse),
+        }
+    }
+
+    /// Fetches a JWT token for the given audience and [`SpiffeId`].
+    ///
+    /// # Arguments
+    ///
+    /// * `audience`  - A list of audiences to include in the JWT token. Cannot be empty nor contain only empty strings.
+    /// * `spiffe_id` - Optional reference [`SpiffeId`] for the token 'sub' claim. If not provided, the Workload API returns the
+    /// default identity,
+    ///
+    /// # Errors
+    ///
+    /// The function returns a variant of [`ClientError`] if there is en error connecting to the Workload API or
+    /// there is a problem processing the response.
+    ///
+    /// IMPORTANT: If there's no registration entries with the requested [`SpiffeId`] mapped to the calling workload,
+    /// it will return a [`ClientError::EmptyResponse`].
+    pub fn fetch_jwt_token<T: AsRef<str> + ToString>(
+        &self,
+        audience: &[T],
+        spiffe_id: Option<&SpiffeId>,
+    ) -> Result<String, ClientError> {
+        let response = self.fetch_jwt(audience, spiffe_id)?;
+        match response.svids.get(DEFAULT_SVID) {
+            Some(r) => Ok(r.svid.to_string()),
+            None => Err(ClientError::EmptyResponse),
+        }
+    }
+}
```

Comment:
Would be very nice to have an abstraction for the `ValidateJWTSVID` call as well but I am happy to add it once this is merged 👌

Reply:
👍